utils.py

```diff
@@ -768,22 +768,15 @@ def convert_to_text_angular(html_data):
     # To remove all tags except those in allowed tag list.
     all_tags = soup.findAll()
     for tag in all_tags:
-        if tag.name not in allowed_tag_list:
+        if tag.name == 'td' and tag.next_sibling:
+            tag.string = tag.string + " "
+        if tag.name == 'div' or tag.name == 'tr':
+            tag.name = 'p'
+        elif tag.name not in allowed_tag_list:
             tag.unwrap()
 
-    # Ensure that if the html content is wrapped in some tag,
-    # if not, wrap it in p tag.
-    tag_list = soup.findAll()
-    if not len(tag_list):
-        start_tag = None
-    else:
-        # This part checks if a html string is not wrapped in any tag. Since it
-        # is not possible to obtain the position of a tag using BeautifulSoup,
-        # exact tag is to be searched in the html string.
-        start_tag = '<' + soup.findAll()[0].name + '>'
-    if not start_tag or html_data.find(start_tag) != 0:
-        html_data = '<p>' + str(soup) + '</p>'
-        soup = bs4.BeautifulSoup(html_data, 'html.parser')
+    # To combine parts after removal of tags.
```

Comment:
If I understand properly, the filtering process above can break the soup into parts, so you need to recombine right?

I suggest making this comment more clear, explaining what the problem is that causes us to need to recombine parts.

Reply:
Done!